global/functions/spawning/fn_spawning_find_valid_position_tracer.sqf

```diff
@@ -77,24 +131,28 @@ while {true} do {
 
 	//If we find a unit, we exit and set the last valid position + which target stopped us.
 	if (!_positionIsValid) exitWith {
-		if (_tracerPosition isEqualTo _tracerStart) then {
-			_finalPosition = [];
-		} else {
-			_finalPosition = _lastTracerPosition;
-		};
+		_finalPosition = _lastValidTracerPosition;
 	};
 
-	//If we reach the destination, we can spawn at the target location without an issue
+	// @dijksterhuis: If we are very close to the end of the projected line
+	// then we need to start spawning in AI at the last known good position.
+	// this means AI spawn one line step further away from the objective,
+	// but this is necessary to avoid AI units spawning at water positions
+	// when setting _finalPosition = _tracerPosition
 	if (_tracerPosition distance2D _tracerEndPos < _stepSize) exitWith {
```

Comment:
Note to self -- the `tracerEndPos` position is never actually checked for whether it is a valid position or not. So it's not really safe to assume it is a good position for spawning.

TODO -- if this moves earlier on then it is possible to check whether the `_tracerEndPos` position is a valid position or not.

```sqf
private _stepSize = ((_tracerPosition distance2D _tracerEndPos) / 10) max 100;
private _newTracerPosition = _tracerPosition getPos [_stepSize, _attackDir];
if (_tracerPosition distance2D _tracerEndPos < _stepSize) then {
    _newTracerPosition = _tracerEndPos;
};
```

Then we just need something like this to break out of the `while` loop
```sqf
if (_tracerPosition == _tracerEndPos) exitWith {
    true;
};
```